### PR TITLE
Update shellcheck version (0.7.1 -> 0.7.2) and fix findings

### PR DESCRIPTION
**Pull request by oscr**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:

Updates shellcheck version 0.7.1 -> 0.7.2 which introduces the following new checks: https://github.com/koalaman/shellcheck/blob/master/CHANGELOG.md#v072---2021-04-19

Fixes the following findings
```
Using shellcheck 0.7.2 docker image.

In ./hack/make-rules/test.sh line 146:
  goflags+=(-v)
  ^-----^ SC2031: goflags was modified in a subshell. That change might be lost.


In ./hack/make-rules/vet.sh line 30:
    goflags+=("${arg}")
    ^-----^ SC2031: goflags was modified in a subshell. That change might be lost.

For more information:
  https://www.shellcheck.net/wiki/SC2031 -- goflags was modified in a subshel...
```

Note this isn't the newest version of shellcheck. There is a version 0.8.0. However in order to avoid having to fix everything at once I made this update first. This will limit the risk and keep this pr smaller.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

File: hack/verify-shellcheck.sh [hunk 1]
```diff
@@ -30,9 +30,9 @@ DOCKER="${DOCKER:-docker}"
 
 # required version for this script, if not installed on the host we will
 # use the official docker image instead. keep this in sync with SHELLCHECK_IMAGE
-SHELLCHECK_VERSION="0.7.1"
-# upstream shellcheck latest stable image as of October 23rd, 2019
-SHELLCHECK_IMAGE="docker.io/koalaman/shellcheck-alpine:v0.7.1@sha256:d6147f30864ddb7c9cf983fc277d345bc315e798e309ddf70062b194843ee252"
+SHELLCHECK_VERSION="0.7.2"
+# upstream shellcheck latest stable image as of November 1st, 2022
```
Comment (spiffxp) [inline hack/verify-shellcheck.sh:34]: As you say in the description, latest stable is now 0.8.0.  I think this comment is unnecessary at this point
```suggestion
```

**Review by spiffxp — APPROVED** [review]
/sig testing
/approve
/lgtm

**Review by spiffxp — CHANGES_REQUESTED** [review]
/lgtm cancel
Sorry mis-clicked

File: hack/verify-shellcheck.sh (hunk 1 above)
Comment (oscr) [reply hack/verify-shellcheck.sh]: You are absolutely right. Deleted that line.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Pull request merged**